lib/mix/lib/mix/tasks/xref.ex

```diff
@@ -189,10 +189,19 @@ defmodule Mix.Tasks.Xref do
     min_cycle_size: :integer
   ]
 
+  @compile_args [
+    "--no-compile",
+    "--no-deps-check",
+    "--no-archives-check",
+    "--no-elixir-version-check"
+  ]
+
```

Comment:
Good find! For the fix, it is probably best to list those options as part of @switches too:

```elixir
    compile: :boolean,
    deps_check: :boolean,
    archives_check: :boolean,
    elixir_version_check: :boolean,
```

Reply:
Move the line that calls compile before we parse them then!